### PR TITLE
Looping Rods: The Return

### PR DESCRIPTION
**Pull request by Ivniinvi**
## About The Pull Request
Rods now have a 1% chance of being a looping rod.

## Why It's Good For The Game
Looping rods are a primordial terror unlike nothing else, that provide an interesting threat dimension to the game. They do have a major impact though, which is why they're so rare. Also, the Feat of Strength achievement is for all intents and purposes impossible without looping rods.

## Testing Photographs and Procedure

<details>

<summary>Screenshots&Videos</summary>

Put screenshots and Videos documenting testing and execution of intended behaviors here

</details>

## Changelog
:cl:
tweak: Immovable rods now have a one percent chance to loop.
/:cl:

### CODE REVIEW

**Comment by EvilDragonfiend** [comment]
I wanted them back as they're interesting. by the way, 1% looks too lower. rods are already low chance.
also, I wonder... does it loop regardless of which side is looping?

File: code/modules/events/immovable_rod.dm [hunk 1]
```diff
@@ -36,7 +36,12 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
 	var/z = pick(SSmapping.levels_by_trait(ZTRAIT_STATION))
 	var/turf/startT = spaceDebrisStartLoc(startside, z)
 	var/turf/endT = spaceDebrisFinishLoc(startside, z)
-	var/atom/rod = new /obj/effect/immovablerod(startT, endT, C.special_target)
+	var/atom/rod
+	if(prob(1))
+		rod = new /obj/effect/immovablerod(startT, endT, C.special_target, TRUE)
+		to_chat(world, "<span class='userdanger'>You feel a wave of dread wash over you.</span>")
```
Comment (Archanial) [inline code/modules/events/immovable_rod.dm]: No

File: code/modules/events/immovable_rod.dm [hunk 2]
```diff
@@ -36,7 +36,11 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
 	var/z = pick(SSmapping.levels_by_trait(ZTRAIT_STATION))
 	var/turf/startT = spaceDebrisStartLoc(startside, z)
 	var/turf/endT = spaceDebrisFinishLoc(startside, z)
-	var/atom/rod = new /obj/effect/immovablerod(startT, endT, C.special_target)
+	var/atom/rod
+	if(prob(1))
+		rod = new /obj/effect/immovablerod(startT, endT, C.special_target, TRUE)
+	else
+		rod = new /obj/effect/immovablerod(startT, endT, C.special_target)
```
Comment (Archanial) [inline code/modules/events/immovable_rod.dm]:
```suggestion
	var/looping = FALSE
	if(prob(1))
		looping = TRUE
	var/obj/effect/immovablerod/rod = new /obj/effect/immovablerod(startT, endT, C.special_target, looping)
```

File: code/modules/events/immovable_rod.dm [hunk 3]
```diff
@@ -93,8 +101,14 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
...
 	var/cur_dist = get_dist(src, destination)
-	if((z != z_original) || (loc == destination) || (FLOOR(cur_dist - previous_distance, 1) > 10))
+	if(z != z_original)
 		qdel(src)
+	if(((loc == destination) || (FLOOR(cur_dist - previous_distance, 1) > 10)) && !looping)
+		qdel(src)
+	if(loc == destination && looping)
+		forceMove(beginning)
+		walk_towards(src, destination, 1)
+		previous_distance = get_dist(src, destination)
```
Comment (Archanial) [inline code/modules/events/immovable_rod.dm]: don't really like logic here, you are checking loc == destination twice and looping twice 🥴

File: code/modules/events/immovable_rod.dm [hunk 4]
```diff
@@ -102,12 +102,13 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
...
-		forceMove(beginning)
-		walk_towards(src, destination, 1)
-		previous_distance = get_dist(src, destination)
+	if(loc == destination)
+		if(looping)
+			forceMove(beginning)
+			walk_towards(src, destination, 1)
+			previous_distance = get_dist(src, destination)
+		else if(FLOOR(cur_dist - previous_distance, 1) > 10)
+			qdel(src)
```
Comment (Archanial) [inline code/modules/events/immovable_rod.dm:111]: shouldn't you early return at the end here?

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Good to have a completely uncounterable rod that cuts the entire station in half.
At least make it only possible when there is a research director in the game, since nobody can stop it apart from the RD.
The achievement is meant to be difficult, the majority of players don't know you can do this so even with an RD in the game this will most likely just exist forever, constantly cutting the station in half until an admin deletes it.

**Comment by Archanial** [comment]
> Good to have a completely uncounterable rod that cuts the entire station in half. At least make it only possible when there is a research director in the game, since nobody can stop it apart from the RD. The achievement is meant to be difficult, the majority of players don't know you can do this so even with an RD in the game this will most likely just exist forever, constantly cutting the station in half until an admin deletes it.

Objectively untrue and massive skills issue.
Just play RD and be gamer.

**Comment by Ivniinvi** [comment]
The rod now checks if your ID is registered as Research Director, instead of your roundstart job, to allow for acting heads and for there to be a counter if there is no RD.

**Comment by Sinestia** [comment]
I swear it's counterable with holobarriers but that might be a schizopost

**Comment by williamcll** [comment]
> 

As far as I know only SM can destroy a rod.

**Comment by Wilchenx** [comment]
> > 
> 
> As far as I know only SM can destroy a rod.

No it just gets stuck in it

**Comment by PowerfulBacon** [comment]
this PR is slowly making less and less sense, to counter the rod being uncounterable now you gain the ability to surplex a rod for just wearing the RDs ID? Also being counterable =/= being balanced, it will still be near impossible to stop and will just be an admin delete thing, reported as a bug and then removed.

**Comment by Ivniinvi** [comment]
> this PR is slowly making less and less sense, to counter the rod being uncounterable now you gain the ability to surplex a rod for just wearing the RDs ID? Also being counterable =/= being balanced, it will still be near impossible to stop and will just be an admin delete thing, reported as a bug and then removed.

I've changed it back to roundstart RD, and I'm going to leave it as spawnable with no RD until either another maintainer weighs in or you and Archanial agree on whether or not this condition should apply.

**Comment by ike709** [comment]
I'm not really a fan of increasing the usage of BYOND `walk()` procs without porting TG's refactor, but the probability on this is so low that it probably doesn't matter

**Comment by PowerfulBacon** [comment]
🗝️

**Comment by ike709** [comment]
A much better rod buff would be to add a generator to predispose it towards the center of the zlevel so it's actually liable to hit the station

🔑

**Pull request closed without merging**